### PR TITLE
1436790/1438462: Part 46: 20 files with MSDN link fixes.

### PR DESCRIPTION
**Pull request by DennisLee-DennisLee**
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

It seems that some links haven't been migrated yet, so I just cleaned them up in other ways (such as removing the "en-us", fixing the link text, or changing them so they have breadcrumbs showing).

### CODE REVIEW

**Comment by mairaw** [comment]
I'm not exactly sure how your process work when you start a new branch for this work, but there's always some extra commits that I don't think would be needed, Do you ever reset/rebase your branch with upstream/master?

File: docs/standard/microservices-architecture/architect-microservice-container-applications/resilient-high-availability-microservices.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Different orchestrators might sound similar, but the diagnostics and health chec
   [*https://docs.microsoft.com/azure/log-analytics/log-analytics-windows-agents*](https://docs.microsoft.com/azure/log-analytics/log-analytics-windows-agents)
 
 - **Logging What You Mean: Using the Semantic Logging Application Block** \
-  [*https://msdn.microsoft.com/library/dn440729(v=pandp.60).aspx*](https://msdn.microsoft.com/library/dn440729(v=pandp.60).aspx)
+  [*https://docs.microsoft.com/previous-versions/msp-n-p/dn440729(v=pandp.60)*](https://docs.microsoft.com/previous-versions/msp-n-p/dn440729(v=pandp.60))
```
Comment (mairaw) [inline docs/standard/microservices-architecture/architect-microservice-container-applications/resilient-high-availability-microservices.md]:
```suggestion
  <https://docs.microsoft.com/previous-versions/msp-n-p/dn440729(v=pandp.60)>
```

File: docs/standard/microservices-architecture/implement-resilient-applications/partial-failure-strategies.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Strategies for dealing with partial failures include the following.
   [*https://docs.microsoft.com/azure/architecture/patterns/category/resiliency*](/azure/architecture/patterns/category/resiliency)
 
 - **Adding Resilience and Optimizing Performance**\
-  [*https://msdn.microsoft.com/library/jj591574.aspx*](https://msdn.microsoft.com/library/jj591574.aspx)
+  [*https://docs.microsoft.com/previous-versions/msp-n-p/jj591574(v=pandp.10)*](https://docs.microsoft.com/previous-versions/msp-n-p/jj591574(v=pandp.10))
```
Comment (mairaw) [inline docs/standard/microservices-architecture/implement-resilient-applications/partial-failure-strategies.md]:
```suggestion
  <https://docs.microsoft.com/previous-versions/msp-n-p/jj591574(v=pandp.10)>
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 3]
```diff
@@ -87,17 +87,17 @@ In more advanced microservices, like when using CQRS approaches, it can be imple
...
 
-In microservices-based architectures, you should choose availability and tolerance, and you should deemphasize strong consistency. Therefore, in most modern microservice-based applications, you usually do not want to use distributed transactions in messaging, as you do when you implement [distributed transactions](https://msdn.microsoft.com/library/ms681205\(v=vs.85\).aspx) based on the Windows Distributed Transaction Coordinator (DTC) with [MSMQ](https://msdn.microsoft.com/library/ms711472\(v=vs.85\).aspx).
+In microservices-based architectures, you should choose availability and tolerance, and you should deemphasize strong consistency. Therefore, in most modern microservice-based applications, you usually do not want to use distributed transactions in messaging, as you do when you implement [distributed transactions](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms681205(v=vs.85)) based on the Windows Distributed Transaction Coordinator (DTC) with [MSMQ](https://msdn.microsoft.com/library/windows/desktop/ms711472(v=vs.85).aspx).
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
In microservices-based architectures, you should choose availability and tolerance, and you should deemphasize strong consistency. Therefore, in most modern microservice-based applications, you usually do not want to use distributed transactions in messaging, as you do when you implement [distributed transactions](https://docs.microsoft.com/previous-versions/windows/desktop/ms681205(v=vs.85)) based on the Windows Distributed Transaction Coordinator (DTC) with [MSMQ](https://msdn.microsoft.com/library/windows/desktop/ms711472(v=vs.85).aspx).
```

File: docs/standard/microservices-architecture/architect-microservice-container-applications/asynchronous-message-based-communication.md [hunk 4]
```diff
@@ -71,7 +71,7 @@ A challenge when implementing an event-driven architecture across multiple micro
 
 - Using [transaction log mining](https://www.scoop.it/t/sql-server-transaction-log-mining).
 
-- Using full [Event Sourcing](https://msdn.microsoft.com/library/dn589792.aspx) pattern.
+- Using full [Event Sourcing](/azure/architecture/patterns/event-sourcing) pattern.
```
Comment (mairaw) [inline docs/standard/microservices-architecture/architect-microservice-container-applications/asynchronous-message-based-communication.md]:
```suggestion
- Using full [Event Sourcing](https://docs.microsoft.com/azure/architecture/patterns/event-sourcing) pattern.
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 5]
```diff
@@ -87,17 +87,17 @@ In more advanced microservices, like when using CQRS approaches, it can be imple
...
 
 Let’s go back to the initial issue and its example. If the service crashes after the database is updated (in this case, right after the line of code with \_context.SaveChangesAsync()), but before the integration event is published, the overall system could become inconsistent. This might be business critical, depending on the specific business operation you are dealing with.
 
 As mentioned earlier in the architecture section, you can have several approaches for dealing with this issue:
 
--   Using the full [Event Sourcing pattern](https://msdn.microsoft.com/library/dn589792.aspx).
+-   Using the full [Event Sourcing pattern](/azure/architecture/patterns/event-sourcing).
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
-   Using the full [Event Sourcing pattern](https://docs.microsoft.com/azure/architecture/patterns/event-sourcing).
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 6]
```diff
@@ -298,7 +298,7 @@ Some message processing is inherently idempotent. For example, if a system gener
 ### Additional resources
 
 -   **Honoring message idempotency** <br/>
-    [*https://msdn.microsoft.com/library/jj591565.aspx#honoring_message_idempotency*](https://msdn.microsoft.com/library/jj591565.aspx)
+    [*https://docs.microsoft.com/previous-versions/msp-n-p/jj591565(v=pandp.10)#honoring-message-idempotency*](https://docs.microsoft.com/previous-versions/msp-n-p/jj591565(v=pandp.10)#honoring-message-idempotency)
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
    <https://docs.microsoft.com/previous-versions/msp-n-p/jj591565(v=pandp.10)#honoring-message-idempotency>
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 7]
```diff
@@ -331,7 +331,7 @@ If the “redelivered” flag is set, the receiver must take that into account,
     [*https://www.enterpriseintegrationpatterns.com/patterns/messaging/PublishSubscribeChannel.html*](https://www.enterpriseintegrationpatterns.com/patterns/messaging/PublishSubscribeChannel.html)
 
 -   **Communicating Between Bounded Contexts** <br/>
-    [*https://msdn.microsoft.com/library/jj591572.aspx*](https://msdn.microsoft.com/library/jj591572.aspx)
+    [*https://docs.microsoft.com/previous-versions/msp-n-p/jj591572(v=pandp.10)*](https://docs.microsoft.com/previous-versions/msp-n-p/jj591572(v=pandp.10))
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
    <https://docs.microsoft.com/previous-versions/msp-n-p/jj591572(v=pandp.10)>
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 8]
```diff
@@ -346,7 +346,7 @@ If the “redelivered” flag is set, the receiver must take that into account,
     [*https://microservices.io/patterns/data/event-sourcing.html*](https://microservices.io/patterns/data/event-sourcing.html)
 
 -   **Introducing Event Sourcing** <br/>
-    [*https://msdn.microsoft.com/library/jj591559.aspx*](https://msdn.microsoft.com/library/jj591559.aspx)
+    [*https://docs.microsoft.com/previous-versions/msp-n-p/jj591559(v=pandp.10)*](https://docs.microsoft.com/previous-versions/msp-n-p/jj591559(v=pandp.10))
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
    <https://docs.microsoft.com/previous-versions/msp-n-p/jj591559(v=pandp.10)>
```

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 9]
```diff
@@ -361,7 +361,7 @@ If the “redelivered” flag is set, the receiver must take that into account,
     [*https://www.quora.com/What-Is-CAP-Theorem-1*](https://www.quora.com/What-Is-CAP-Theorem-1)
 
 -   **Data Consistency Primer** <br/>
-    [*https://msdn.microsoft.com/library/dn589800.aspx*](https://msdn.microsoft.com/library/dn589800.aspx)
+    [*https://docs.microsoft.com/previous-versions/msp-n-p/dn589800(v=pandp.10)*](https://docs.microsoft.com/previous-versions/msp-n-p/dn589800(v=pandp.10))
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]:
```suggestion
    <https://docs.microsoft.com/previous-versions/msp-n-p/dn589800(v=pandp.10)>
```

File: docs/standard/modernize-with-azure-and-containers/modernize-existing-apps-to-cloud-optimized/build-resilient-services-ready-for-the-cloud-embrace-transient-failures-in-the-cloud.md [hunk 10]
```diff
@@ -43,11 +43,11 @@ To learn how to implement strategies for handling partial failures in the cloud,
 
 -   **Entity Framework connection resiliency and retry logic (version 6 and later)**
 
-    [https://msdn.microsoft.com/library/dn456835(v=vs.113).aspx](https://msdn.microsoft.com/library/dn456835(v=vs.113).aspx)
+    [https://docs.microsoft.com/ef/ef6/fundamentals/connection-resiliency/retry-logic](/ef/ef6/fundamentals/connection-resiliency/retry-logic)
 
 -   **The Transient Fault Handling Application Block**
 
--   [https://msdn.microsoft.com/library/hh680934(v=pandp.50).aspx](https://msdn.microsoft.com/library/hh680934(v=pandp.50).aspx)
+-   [https://docs.microsoft.com/previous-versions/msp-n-p/hh680934(v=pandp.50)](https://docs.microsoft.com/previous-versions/msp-n-p/hh680934(v=pandp.50))
```
Comment (mairaw) [inline docs/standard/modernize-with-azure-and-containers/modernize-existing-apps-to-cloud-optimized/build-resilient-services-ready-for-the-cloud-embrace-transient-failures-in-the-cloud.md]:
```suggestion
-   <https://docs.microsoft.com/previous-versions/msp-n-p/hh680934(v=pandp.50)>
```

File: docs/standard/parallel-programming/potential-pitfalls-in-data-and-task-parallelism.md [hunk 11]
```diff
@@ -46,10 +46,10 @@ In many cases, <xref:System.Threading.Tasks.Parallel.For%2A?displayProperty=name
...
+ Some technologies, for example, COM interoperability for Single-Threaded Apartment (STA) components, Windows Forms, and Windows Presentation Foundation (WPF), impose thread affinity restrictions that require code to run on a specific thread. For example, in both Windows Forms and WPF, a control can only be accessed on the thread on which it was created. This means, for example, that you cannot update a list control from a parallel loop unless you configure the thread scheduler to schedule work only on the UI thread. For more information, see [How to: Schedule Work on the User Interface (UI) Thread](xref:System.Threading.Tasks.TaskScheduler#specifying-a-synchronization-context).
```
Comment (mairaw) [inline docs/standard/parallel-programming/potential-pitfalls-in-data-and-task-parallelism.md]:
```suggestion
 Some technologies, for example, COM interoperability for Single-Threaded Apartment (STA) components, Windows Forms, and Windows Presentation Foundation (WPF), impose thread affinity restrictions that require code to run on a specific thread. For example, in both Windows Forms and WPF, a control can only be accessed on the thread on which it was created. This means, for example, that you cannot update a list control from a parallel loop unless you configure the thread scheduler to schedule work only on the UI thread. For more information, see [Specifying a synchronization context](xref:System.Threading.Tasks.TaskScheduler#specifying-a-synchronization-context).  
```

File: docs/standard/security/cryptographic-services.md [hunk 12]
```diff
@@ -190,7 +190,7 @@ ms.author: "mairaw"
...
   
-     Unfortunately, this method does not establish the authenticity of the sender. Anyone can impersonate Alice and send a message to Bob. They can use the same hash algorithm to sign their message, and all Bob can determine is that the message matches its signature. This is one form of a man-in-the-middle attack. See [NIB: Cryptography Next Generation (CNG) Secure Communication Example](https://msdn.microsoft.com/library/8048e94e-054a-417b-87c6-4f5e26710e6e) for more information.  
+     Unfortunately, this method does not establish the authenticity of the sender. Anyone can impersonate Alice and send a message to Bob. They can use the same hash algorithm to sign their message, and all Bob can determine is that the message matches its signature. This is one form of a man-in-the-middle attack. See [Cryptography Next Generation (CNG) Secure Communication Example](https://docs.microsoft.com/previous-versions/cc488018(v=vs.100)) for more information.
```
Comment (mairaw) [inline docs/standard/security/cryptographic-services.md]:
```suggestion
     Unfortunately, this method does not establish the authenticity of the sender. Anyone can impersonate Alice and send a message to Bob. They can use the same hash algorithm to sign their message, and all Bob can determine is that the message matches its signature. This is one form of a man-in-the-middle attack. For more information, see [Cryptography Next Generation (CNG) Secure Communication Example](https://docs.microsoft.com/previous-versions/cc488018(v=vs.100)).  
```

File: docs/standard/security/how-to-encrypt-xml-elements-with-x-509-certificates.md [hunk 13]
```diff
@@ -22,15 +22,15 @@ You can use the classes in the <xref:System.Security.Cryptography.Xml> namespace
...
+ This example encrypts an XML element using two keys.  It generates a test X.509 certificate using the [Certificate Creation Tool (Makecert.exe)](/windows/desktop/SecCrypto/makecert) and saves the certificate to a certificate store.  The example then programmatically retrieves the certificate and uses it to encrypt an XML element using the <xref:System.Security.Cryptography.Xml.EncryptedXml.Encrypt%2A> method.  Internally, the <xref:System.Security.Cryptography.Xml.EncryptedXml.Encrypt%2A> method creates a separate session key and uses it to encrypt the XML document. This method encrypts the session key and saves it along with the encrypted XML within a new <`EncryptedData`> element.
```
Comment (mairaw) [inline docs/standard/security/how-to-encrypt-xml-elements-with-x-509-certificates.md]:
```suggestion
 This example encrypts an XML element using two keys. It generates a test X.509 certificate using the [Certificate Creation Tool (Makecert.exe)](/windows/desktop/SecCrypto/makecert) and saves the certificate to a certificate store. The example then programmatically retrieves the certificate and uses it to encrypt an XML element using the <xref:System.Security.Cryptography.Xml.EncryptedXml.Encrypt%2A> method. Internally, the <xref:System.Security.Cryptography.Xml.EncryptedXml.Encrypt%2A> method creates a separate session key and uses it to encrypt the XML document. This method encrypts the session key and saves it along with the encrypted XML within a new <`EncryptedData`> element.  
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thank you @DennisLee-DennisLee. A few comments before we merge...

**Comment by DennisLee-DennisLee** [comment]
@mairaw Sorry, I've noticed those commits building up in recent days. I think the problem stems from getting a branch corrupted while I was working on the 38th PR. I do pull from upstream/master, but I've never used rebase (and generally try to avoid reset).

```output
$ git branch
warning: ignoring broken ref refs/heads/v-dele-1438462-038
  master
  v-dele-1438462-046
* v-dele-1438462-047

```

**Comment by mairaw** [comment]
Ok @DennisLee-DennisLee. I think it would be good to do a git reset like this just to start with some clean PRs:

`git reset --hard upstream/master`

**Comment by mairaw** [comment]
@mvelosop FYI, some link updates on some of the e-book content.

**Review by mairaw — APPROVED** [review]
Thanks!

**Pull request merged**